src/language/I18n.ts

```diff
@@ -0,0 +1,35 @@
+import i18n from 'i18next';
+import { initReactI18next } from 'react-i18next';
+import LanguageDetector from 'i18next-browser-languagedetector';
+
+import { generalConfigService } from '../storage/GeneralConfigService';
+import translations from './translations.json';
+
+let savedLanguage: string = '';
+
+const getLanguageFromDB = async () => {
+  const language = await generalConfigService.getLanguage();
+  savedLanguage = language;
+};
+
+getLanguageFromDB();
+
+i18n
+  // detect user language
+  // learn more: https://github.com/i18next/i18next-browser-languageDetector
+  .use(LanguageDetector)
+  // pass the i18n instance to react-i18next.
+  .use(initReactI18next)
+  // init i18next
+  // for all options read: https://www.i18next.com/overview/configuration-options
+  .init({
+    debug: true,
+    lng: savedLanguage,
```

Comment:
```suggestion
    lng: await generalConfigService.getLanguage(),
```

I wonder if it's possible to call it directly this way and thus avoiding the init method `getLanguageFromDB` altogether

Reply:
While simply using the suggested change is not possible, as `i18n.init()` is not an `async` funciton, but I've managed to use a proper implementation with using `LanguageDetector` Instance. 
```
const languageDetector = new LanguageDetector(async () => {
  return await generalConfigService.getLanguage();
});

i18n
  // detect user language
  // learn more: https://github.com/i18next/i18next-browser-languageDetector
  .use(languageDetector)
  // pass the i18n instance to react-i18next.
  .use(initReactI18next)
  // init i18next
  // for all options read: https://www.i18next.com/overview/configuration-options
  .init({
    debug: true,
    fallbackLng: 'enUS',
    interpolation: {
      escapeValue: false, // not needed for react as it escapes by default
    },
    resources: translations,
  });

export default i18n;
```